omaha/core/scheduler.h

```diff
@@ -20,45 +20,87 @@
 #define OMAHA_CORE_SCHEDULER_H__
 
 #include <windows.h>
-#include <atlstr.h>
+#include <functional>
+#include <list>
 #include <memory>
 
 #include "base/basictypes.h"
+#include "omaha/base/highres_timer-win32.h"
+#include "omaha/base/queue_timer.h"
 
 namespace omaha {
 
-class Core;
-class HighresTimer;
-class QueueTimer;
+using ScheduledWork = std::function<void(HighresTimer*)>;
 
 class Scheduler {
  public:
-  explicit Scheduler(const Core& core);
+  explicit Scheduler();
   ~Scheduler();
 
-  // Starts the scheduler.
-  HRESULT Initialize();
+  // Starts the scheduler that executes |work| when |interval| (ms) elapses,
+  // after an initial delay of |start_delay| (ms).
+  // Default debug timer to false.
+  HRESULT Start(int start_delay,
+                int interval,
+                ScheduledWork work,
+                bool has_debug_timer = false);
+
+  // Starts the scheduler that executes |work| with regular |interval| (ms)
+  HRESULT Start(int interval, ScheduledWork work, bool has_debug_timer = false);
 
  private:
-  static void TimerCallback(QueueTimer* timer);
-  void HandleCallback(QueueTimer* timer);
-  HRESULT ScheduleUpdateTimer(int interval_ms);
-  HRESULT ScheduleCodeRedTimer(int interval_ms);
+  // Represents a unit of work that scheduler executes.
+  class SchedulerItem {
+  public:
+    SchedulerItem(HANDLE timer_queue,
+                  int start_delay,
+                  int interval,
+                  bool has_debug_timer,
+                  ScheduledWork work_fn);
+
+    ~SchedulerItem();
+
+    HighresTimer* GetDebugTimer() {
+      return debug_timer_ ? debug_timer_.get() : nullptr;
+    }
+
+    int GetIntervalMs() const { return interval_ms_; }
```

Comment:
naming convention for accessors like this and the one above is naming them as variables, such as interval_ms().

Reply:
Done